### ISSUE TITLE
False Negative: RedundantModifierCheck

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11268]
I have read check documentation: https://checkstyle.sourceforge.io/config_modifier.html#RedundantModifier
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RedundantModifier"/>
    </module>
</module>

/var/tmp $ cat Test.java
package tmp;

public interface Test {
    public static interface foo { // violation for 'static', no violation for 'public'
    }

    public static enum someEnum { // violation for both
    }

    public static class someClass { // violation for both
    }

    public static record someRecord() { // violation for none, will be fixed in #11259
    }

    public static @interface someAnnInterface { // violation for none
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:4:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:7:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:7:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:10:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:10:12: Redundant 'static' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 5 errors.

```
---

**Describe what you expect in detail.**
```
Starting audit...
[ERROR] /var/tmp/Test.java:4:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:4:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:7:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:7:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:10:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:10:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:13:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:13:12: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:16:5: Redundant 'public' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:16:12: Redundant 'static' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 10 errors.
```
Every member class or interface declaration in the body of an interface declaration is implicitly public and static. See [JLS](https://docs.oracle.com/javase/specs/jls/se17/html/jls-9.html#jls-9.5) for more info. Comments in code specify the current behavior.

**Comment by Vyom-Yadav** [comment]
I will send the fix once approved.

**Comment by nmancus1** [comment]
```
    public static interface foo { // violation for 'static', no violation for 'public'
    }
```
Please explain why `public` should not be a violation here.

**Comment by nmancus1** [comment]
@Vyom-Yadav issue is approved, please send PR.

**Comment by romani** [comment]
fix is merged
